src/import/crd.ts

```diff
@@ -102,49 +105,29 @@ export class CustomResourceDefinition {
 }
 
 export class ImportCustomResourceDefinition extends ImportBase {
-  public static async match(importSpec: ImportSpec): Promise<undefined | ManifestObjectDefinition[]> {
+  public static async fromSpec(importSpec: ImportSpec): Promise<ImportCustomResourceDefinition> {
     const { source } = importSpec;
     const manifest = await download(source);
-    return yaml.parseAllDocuments(manifest).map((doc: yaml.Document) => doc.toJSON());
+    return new ImportCustomResourceDefinition(safeParseCrds(manifest));
   }
 
   private readonly groups: Record<string, CustomResourceDefinition[]> = { };
 
-  constructor(manifest: ManifestObjectDefinition[]) {
+  private constructor(manifest: ManifestObjectDefinition[]) {
```

Comment:
Made this private so that this object cannot be created without input validation.